subgraphs/gmx/src/mappings/vault.ts

```diff
@@ -0,0 +1,457 @@
+  const outputTokenAmountUSD = convertTokenToDecimal(
+    event.params.amountOutAfterFees,
+    outputToken.decimals
+  ).times(outputToken.lastPriceUSD!);
+
+  createSwap(
+    event,
+    event.params.account,
+    event.params.tokenIn,
+    event.params.amountIn,
+    inputTokenAmountUSD,
+    event.params.tokenOut,
+    event.params.amountOutAfterFees,
+    outputTokenAmountUSD
+  );
+
+  updateTempUsageMetrics(
+    event,
+    event.params.account,
+    EventType.Swap,
+    INT_ZERO,
+    null
+  );
+}
+
+export function handleIncreasePosition(event: IncreasePosition): void {
+  handleUpdatePositionEvent(
+    event,
+    event.params.key,
+    event.params.account,
+    event.params.collateralToken,
+    event.params.collateralDelta,
+    event.params.indexToken,
+    event.params.sizeDelta,
+    event.params.price,
+    event.params.fee,
+    event.params.isLong,
+    EventType.CollateralIn,
+    BIGINT_ZERO
+  );
+}
+
+export function handleDecreasePosition(event: DecreasePosition): void {
+  handleUpdatePositionEvent(
+    event,
+    event.params.key,
+    event.params.account,
+    event.params.collateralToken,
+    event.params.collateralDelta,
+    event.params.indexToken,
+    event.params.sizeDelta,
+    event.params.price,
+    event.params.fee,
+    event.params.isLong,
+    EventType.CollateralOut,
+    BIGINT_ZERO
+  );
+}
+
+export function handleLiquidatePosition(event: LiquidatePosition): void {
+  handleUpdatePositionEvent(
+    event,
+    event.params.key,
+    event.params.account,
+    event.params.collateralToken,
+    event.params.collateral,
+    event.params.indexToken,
+    event.params.size,
+    event.params.markPrice,
+    BIGINT_ZERO,
+    event.params.isLong,
+    EventType.Liquidated,
+    event.params.realisedPnl
+  );
+}
+
+export function handleUpdatePositionEvent(
+  event: ethereum.Event,
+  positionKey: Bytes,
+  accountAddress: Address,
+  collateralTokenAddress: Address,
+  collateralDelta: BigInt,
+  indexTokenAddress: Address,
+  sizeDelta: BigInt,
+  indexTokenPrice: BigInt,
+  fee: BigInt,
+  isLong: boolean,
+  eventType: EventType,
+  liqudateProfit: BigInt
+): void {
+  takeSnapshots(event);
+
+  const account = getOrCreateAccount(event, accountAddress);
+  incrementAccountEventCount(event, account, eventType, sizeDelta);
+  incrementProtocolEventCount(event, eventType, sizeDelta);
+
+  const indexToken = getOrCreateToken(event, indexTokenAddress);
+  updateTokenPrice(
+    event,
+    indexToken,
+    convertPriceToBigDecimal(indexTokenPrice)
+  );
+  const sizeUSDDelta = convertPriceToBigDecimal(sizeDelta);
+  const collateralToken = getOrCreateToken(event, collateralTokenAddress);
+  const collateralUSDDelta = convertPriceToBigDecimal(collateralDelta);
+  let collateralTokenAmountDelta = BIGINT_ZERO;
+  if (
+    collateralToken.lastPriceUSD &&
+    collateralToken.lastPriceUSD! > BIGDECIMAL_ZERO
+  ) {
+    collateralTokenAmountDelta = bigDecimalToBigInt(
```

Comment:
Why use bigint for USD amount?

Reply:
This is the native amount, it is found through the USD value